src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_image_wrap.h

```diff
@@ -520,6 +538,12 @@ namespace sirf {
 			dim[2] = im.getMatrixSizeZ();
 			dim[3] = im.getNumberOfChannels();
 		}
+        template<typename T>
+		void get_data_type_(const ISMRMRD::Image<T>* ptr_im, ISMRMRD::ISMRMRD_DataTypes* data_type_ptr) const
+		{
+			const ISMRMRD::Image<T>& im = *static_cast<const ISMRMRD::Image<T>*>(ptr_im);
```

Comment:
very complicated way to write the cast... In fact, it looks like there's no cast at all...